cmd/kubelet/app/server.go

```diff
@@ -906,6 +916,9 @@ func buildKubeletClientConfig(ctx context.Context, s *options.KubeletServer, nod
 			utilnet.CloseIdleConnectionsFor(clientConfig.Transport)
 		}
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.KubeletTracing) && s.KubeletConfiguration.Tracing != nil {
```

Comment:
It's confusing that we're checking for a non-nil `s.KubeletConfiguration.Tracing` here, but not passing it to WrapperFor. Do we need the nil check any more, or can we assume tp will be a non-nil no-op tracing provider if no tracing is configured?

Reply:
I think the purpose of the if condition judgment here is that if s.KubeletConfiguration.Tracing is nil, it is meaningless to call WrapperFor(tp).

Reply:
right, we don't need the nil check there, it will be the no-op. updated
(https://github.com/kubernetes/kubernetes/blob/a6544b59c7cfa6d7673b74bd02c79214b932d22d/cmd/kubelet/app/server.go#L1315-#L1331)